controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -237,6 +238,33 @@ public String getName() {
                                 .thenCompose(deleted -> storeHelper.removeEntry(SCOPES_TABLE, scopeName, context.getRequestId()));
     }
 
+    @Override
+    public CompletableFuture<Void> deleteScopeRecursive(OperationContext context) {
+        Preconditions.checkNotNull(context, "Operation context cannot be null");
+        CompletableFuture<String> streamsInScopeTableNameFuture = getStreamsInScopeTableName(true, context);
+        CompletableFuture<String> rgsInScopeTableNameFuture = getReaderGroupsInScopeTableName(context);
+        CompletableFuture<String> kvtsInScopeTableNameFuture = getKVTablesInScopeTableName(context);
+        CompletableFuture<List<String>> streamTagsInScopeTableNameFuture = getAllStreamTagsInScopeTableNames(context);
+        return CompletableFuture.allOf(streamsInScopeTableNameFuture, rgsInScopeTableNameFuture, kvtsInScopeTableNameFuture, streamTagsInScopeTableNameFuture)
+                .thenCompose(x -> {
+                    String streamsInScopeTableName = streamsInScopeTableNameFuture.join();
+                    String kvtsInScopeTableName = kvtsInScopeTableNameFuture.join();
+                    String rgsInScopeTableName = rgsInScopeTableNameFuture.join();
+                    List<String> streamTagsInScopeTableNames = streamTagsInScopeTableNameFuture.join();
+                    val deleteTagTablesFut = streamTagsInScopeTableNames.stream()
+                            .map(tableName -> storeHelper.deleteTable(tableName, false, context.getRequestId()))
+                            .collect(Collectors.toList());
+                    return CompletableFuture.allOf(
+                            storeHelper.deleteTable(streamsInScopeTableName, true, context.getRequestId()),
+                                    storeHelper.deleteTable(kvtsInScopeTableName, true, context.getRequestId()),
+                                    storeHelper.deleteTable(rgsInScopeTableName, true, context.getRequestId()),
+                                    Futures.allOf(deleteTagTablesFut))
+                            .thenAccept(v -> log.debug("tables deleted {} {} {}", streamsInScopeTableName,
+                                    kvtsInScopeTableName, rgsInScopeTableName));
+                }).thenCompose(deleted -> storeHelper.removeEntry(DELETING_SCOPES_TABLE, scopeName, context.getRequestId()))
```

Comment:
Please remove code duplication between deleteScope() and deleteScopeRecursive().

Reply:
handled by extracting the common part to a new method and the call is made from both `deleteScope()` and `deleteScopeRecursive()`